### PR TITLE
Increase test coverage on back-end

### PR DESCRIPTION
**Pull request by davidefalcone1**
# Description

Add test files and some unit tests in existing ones in order to increase the global coverage on the back-end

### CODE REVIEW

**Review by palexster — APPROVED** [review]

File: operators/pkg/instance-creation/exposition_test.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	},
+	}
+
+	service := ForgeService(name, namespace, ownerRef)
+
+	assert.Equal(t, service.ObjectMeta.Name, name+"-svc")
+	assert.Equal(t, service.ObjectMeta.Namespace, namespace)
+	assert.Equal(t, service.OwnerReferences[0].APIVersion, "crownlabs.polito.it/v1alpha2")
+	assert.Equal(t, service.OwnerReferences[0].Kind, "Instance")
+	assert.Equal(t, service.OwnerReferences[0].Name, "Test1")
```
Comment (giorio94) [inline operators/pkg/instance-creation/exposition_test.go:31]: It should also be tested whether the spec of the service is correct

File: operators/pkg/instance-creation/exposition_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	assert.Equal(t, ingress.ObjectMeta.Annotations["nginx.ingress.kubernetes.io/configuration-snippet"], `sub_filter '<head>' '<head> <base href="https://$host/`+urlUUID+`/index.html">';`)
+	assert.Equal(t, ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Path, "/"+urlUUID+"(/|$)(.*)")
+	assert.Equal(t, ingress.ObjectMeta.Annotations["crownlabs.polito.it/probe-url"], "https://"+url)
+	assert.Equal(t, ingress.Spec.TLS[0].Hosts[0], websiteBaseUrl)
+	assert.Equal(t, ingress.Spec.Rules[0].Host, websiteBaseUrl)
+	assert.Equal(t, ingress.OwnerReferences[0].APIVersion, "crownlabs.polito.it/v1alpha2")
+	assert.Equal(t, ingress.OwnerReferences[0].Kind, "Instance")
+	assert.Equal(t, ingress.OwnerReferences[0].Name, "Test1")
+	assert.Equal(t, ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Backend.Service.Name, svc.Name)
+	assert.Equal(t, ingress.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Backend.Service.Port.Number, svc.Spec.Ports[0].TargetPort.IntVal)
```
Comment (giorio94) [inline operators/pkg/instance-creation/exposition_test.go:79]: These tests seem duplicated to me. Please order the checks to prevent this issue

**Review by giorio94 — COMMENTED** [review]
Thanks @davidefalcone1 for this. Just a couple of comments inline

File: operators/pkg/instance-creation/exposition_test.go (hunk 1 above)
Comment (davidefalcone1) [reply operators/pkg/instance-creation/exposition_test.go:31]: Done, thanks

File: operators/pkg/instance-creation/exposition_test.go (hunk 2 above)
Comment (davidefalcone1) [reply operators/pkg/instance-creation/exposition_test.go]: You are right, thank you

**Review by giorio94 — APPROVED** [review]
/lgtm

Please, rebase you changes and we can merge

**Pull request merged**